app/jobs/finish_sync_job.rb

```diff
@@ -1,25 +1,29 @@
-class DataWarehouseSyncJob < ApplicationJob
-  def perform
-    return if nothing_to_sync?
+class FinishSyncJob < ApplicationJob
+  attr_accessor :sync
 
-    SlackBot.post('sync starting')
-    update_finished_imports
+  def perform(sync_id)
+    @sync = Sync.find_by id: sync_id
+    return unless sync
```

Comment:
If this is a common pattern, we could add a `rescue_from` to `ApplicationJob`:

```ruby
class ApplicationJob < ActiveJob::Base
  rescue_from ActiveRecord::RecordNotFound do |record_not_found|
    # log if needed
  end
end

class FinishSyncJob < ApplicationJob
  ...

  def perform(sync_id)
    @sync = Sync.find(sync_id) # just find instead of #find_by(...)

    ...
  end
end
```